Magento/magento_readme.md

```diff
@@ -0,0 +1,58 @@
+
+## Solution architecture
+
+The solution architecture for Magento running on Azure is shown in the following diagram.
+
+![Magento solution architecture](images/magento-architecture.png)
+
+The Azure services involved in this solution architecture are described below.
+
+* [AKS](https://azure.microsoft.com/services/kubernetes-service/) deploys the Kubernetes cluster of Varnish, Magento, Redis, and [Elasticsearch](https://www.elastic.co/elasticsearch/) in different pods.
+* AKS creates a [virtual network](https://azure.microsoft.com/services/virtual-network/) to deploy the agent nodes. Create the virtual network in advance to set up subnet configuration, private link, and egress restriction.
+* [Varnish](https://varnish-cache.org/intro/index.html#intro) installs in front of the HTTP servers to act as a full-page cache.
+* [Azure Database for MySQL](https://azure.microsoft.com/services/mysql/) stores transaction data such as orders and catalogs. Using version 8.0 is recommended.
+* [Azure Files Premium](https://azure.microsoft.com/services/storage/files/) or an equivalent network-attached storage (NAS) system stores media files such as product images. Magento needs a Kubernetes-compatible file system such as Azure Files Premium or [Azure NetApp Files](https://azure.microsoft.com/services/netapp/), which can mount a volume in *ReadWriteMany* mode.
+* A [content delivery network (CDN)](https://azure.microsoft.com/services/cdn/) serves static content such as CSS, JavaScript, and images. Serving content through a CDN minimizes network latency between users and the datacenter. A CDN can remove significant load from NAS by caching and serving static content.
+* [Redis](https://redis.io/) stores session data. Hosting Redis on containers is recommended for performance reasons.
+* AKS uses an Azure [Active Directory (Azure AD)](https://azure.microsoft.com/services/active-directory/) identity to create and manage other Azure resources like Azure load balancers, user authentication, role-based access control, and managed identity.
+* [Azure Container Registry](https://azure.microsoft.com/services/container-registry/) stores the private [Docker](https://www.docker.com/) images that are deployed to the AKS cluster. You can use other container registries like Docker Hub. Note that the default Magento install writes some secrets to the image.
+* [Azure Monitor](https://azure.microsoft.com/services/monitor/) collects and stores metrics and logs, including Azure service platform metrics and application telemetry. Azure Monitor integrates with AKS to collect controller, node, and container metrics, and container and master node logs.
+
+## How to deploy the ARM template
+
+To deploy this ARM template, open the Azure CLI, and then run the following command:
+
+```azurecli-interactive
+az deployment group create --resource-group <ResourceGroupName> --template-file azuredeploy.json
+```

```

Comment:
Add a link in case they run into issues on how to troubleshoot it . You can point them to https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/common-deployment-errors

Reply:
Added to the new PR